Implement subgroups endpoint

Rimgelair opened the pull request:
Fix https://github.com/ubisoft/NGitLab/issues/295

Rimgelair committed:
Added Subgroup Methods to Interface

Rimgelair committed:
Implement Two Subgroup Requests

- Implemented GetSubgroupsByIdAsync
- Implemented GetSubgroupsByFullPathAsync
- updated the CreateGetUrl to support the two new methods

Rimgelair committed:
Add test for new endpoint

Rimgelair committed:
Add Subgroups Endpoint to Mock

- add the subgroups endpoint to the mock
- add test for the mock

Rimgelair committed:
Clean Up Changes in GroupsClients

- renamed parameter on createGetUrl to be more clear
- removed unecessary string concat on the subgroup endpoint methods

Rimgelair committed:
Merge branch 'main' into implement-subgroups-endpoint

meziantou commented on NGitLab/Impl/GroupsClient.cs:
```diff
@@ -112,6 +117,18 @@ public Task<Group> GetByFullPathAsync(string fullPath, CancellationToken cancell
             return _api.Get().ToAsync<Group>(Url + "/" + Uri.EscapeDataString(fullPath), cancellationToken);
         }
 
+        public GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query)
```
I think query should allow null and be optional. Something like 

```suggestion
        public GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query = null)
```

Rimgelair replied:
I agree I have made the changes.

meziantou commented on NGitLab/IGroupsClient.cs:
```diff
@@ -36,6 +36,10 @@ public interface IGroupsClient
 
         Task<Group> GetByFullPathAsync(string fullPath, CancellationToken cancellationToken = default);
 
+        GitLabCollectionResponse<Group> GetSubgroupsByIdAsync(int id, GroupQuery query);
```
You can't reuse `GroupQuery` here. The available options are not the same (e.g. `top_level_only`). You must create new class `SubgroupQuery`.

Rimgelair replied:
That was an oversight haha, I only checked the GroupQuery class when I should have cross referenced with he docs. I have made the change.

Rimgelair committed:
Apply PR Comments

- Both Request have use the subgroup query
- Sugroupquery is now a optional param that defaults to null
- update the mock and it tests
- update tests

Rimgelair committed:
Merge branch 'implement-subgroups-endpoint' of https://github.com/Rimgelair/NGitLab into implement-subgroups-endpoint

Rimgelair committed:
Merge branch 'main' into implement-subgroups-endpoint

meziantou commented on NGitLab/Models/SubgroupQuery.cs:
```diff
@@ -0,0 +1,63 @@
+        /// Skip the group IDs passed
+        /// (optional)
+        /// </summary>
+        public int[] SkipGroups;
```
Use properties everywhere (using fields is _legacy_)

```suggestion
        public int[] SkipGroups { get; set; }
```

Rimgelair replied:
Done 😄

Rimgelair committed:
Merge branch 'main' into implement-subgroups-endpoint

Rimgelair committed:
Address PR Comments

- Subgroup query fields are now all properties
